salt/modules/btrfs.py

```diff
@@ -673,3 +673,465 @@ def properties(obj, type=None, set=None):
+        salt '*' btrfs.subvolume_delete /var/volumes/tmp commit=after
+
+    '''
+    if not name and not (names and type(names) is list):
+        raise CommandExecutionError('Provide a value for the name parameter')
+
+    if commit and commit not in ('after', 'each'):
+        raise CommandExecutionError('Value for commit not recognized')
+
+    cmd = ['btrfs', 'subvolume', 'delete']
+    if commit == 'after':
+        cmd.append('--commit-after')
+    if commit == 'each':
+        cmd.append('--commit-each')
+    if name:
+        cmd.append(name)
+    if type(names) is list:
+        cmd.extend(names)
+
+    res = __salt__['cmd.run_all'](cmd)
+    salt.utils.fsutils._verify_run(res)
+    return True
+
+
+def subvolume_find_new(name, last_gen):
+    '''
+    List the recently modified files in a subvolume
+
+    name
+        Name of the subvolume
+
+    last_gen
+        Last transid marker from where to compare
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' btrfs.subvolume_find_new /var/volumes/tmp 1024
+
+    '''
+    cmd = ['btrfs', 'subvolume', 'find-new', name, last_gen]
+
+    res = __salt__['cmd.run_all'](cmd)
+    salt.utils.fsutils._verify_run(res)
+
+    lines = res['stdout'].splitlines()
+    # Filenames are at the end of each inode line
+    files = [l.split()[-1] for l in lines if l.startswith('inode')]
+    # The last transid is in the last line
+    transid = lines[-1].split()[-1]
+    return {
+        'files': files,
+        'transid': transid,
+    }
+
+
+def subvolume_get_default(path):
+    '''
+    Get the default subvolume of the filesystem path
+
+    path
+        Mount point for the subvolume
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' btrfs.subvolume_get_default /var/volumes/tmp
+
+    '''
+    cmd = ['btrfs', 'subvolume', 'get-default', path]
+
+    res = __salt__['cmd.run_all'](cmd)
+    salt.utils.fsutils._verify_run(res)
+
+    line = res['stdout'].strip()
+    # The ID is the second parameter, and the name the last one, or
+    # '(FS_TREE)'
+    #
+    # When the default one is set:
+    # ID 5 (FS_TREE)
+    #
+    # When we manually set a different one (var):
+    # ID 257 gen 8 top level 5 path var
+    #
+    id_ = line.split()[1]
+    name = line.split()[-1]
+    return {
+        'id': id_,
+        'name': name,
+    }
+
+
+def _pop(line, key, use_rest):
+    '''
+    Helper for the line parser.
+
+    If key is a prefix of line, will remove ir from the line and will
+    extract the value (space separation), and the rest of the line.
+
+    If use_rest is True, the value will be the rest of the line.
+
+    Return a tuple with the value and the rest of the line.
+    '''
+    value = None
+    if line.startswith(key):
+        line = line[len(key):]
+        line = line.strip()
+        if use_rest:
+            value = line
+            line = ''
+        else:
+            value, line = line.split(' ', 1)
+    return (value, line.strip())
```

Comment:
Extra parenthesis

Reply:
Or so:
```python
value = None
if line.startswith(key):
    line = line[len(key):].strip()
    value, line = use_rest and (line, '') or line.split(' ', 1)
return value, line.strip()

Reply:
Why to obfuscate the code with `and`s and `or`s like if this is pre-Python 2.4?

I removed the extra line with `.strip()` and removed the optional parents (I usually like to be explicit when I return a tuple, but I guess that here is clear enough)